### ISSUE TITLE
CassandraSpanConsumer should not write the remoteEndpoint's span name to the span_by_service table

### ISSUE THREAD

**Issue by drolando** [openzipkin/zipkin#2337]
https://github.com/openzipkin/zipkin/blob/master/zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/CassandraSpanConsumer.java#L96,L97,L98,L99

On the server I use a low cardinality span name (like `get /v1/records/user/{user_id}`). That's possible since pyramid matches the incoming url against all my possible routes and allows me to retrieve the route pattern or the endpoint name.
On the client side I don't have access to either of those, I only have a raw url. So I put that one as span name since it'd be useful if the server is not instrumented. (i.e. `get /v1/records/user/12345678`)

However the `span_by_service` table contains both client and server span names, which means the dropdown in the search page is polluted and unusable. The best part is, filtering on any of the client span names doesn't return anything since the `/spans` endpoint only cares about `localEndpoint`. So I don't see absolutely any point in writing those names.

Can we stop writing them? Or at least add a configuration flag that allows me to turn that off?

cc @michaelsembwever @adriancole

**Comment by codefromthecrypt** [comment]
I think this is mostly a duplicate of https://github.com/openzipkin/zipkin/issues/1794 as it applies to everything.

**Comment by codefromthecrypt** [comment]
ps I was on holiday when you replied. one thing about your specific question is basically you want something different. you want to conditionally index span names depending on the type of the span. putting a path in any type of span is incorrect. the default choice should be low cardinality worst case http method. you can always set a tag for the path. setting high cardinality for a name will bust other things and I don't expect us to conditionally try to control span name indexing.

in other words you are asking to conditionally control span name, not service name indexing. correct me if I am wrong.

**Comment by codefromthecrypt** [comment]
closest to what I think we need here is actually to clean up data in general. For example, if the client span name is high cardinality delete it, as instrumentation should not add high cardinality names. There are a few projects with data cleaning aims. otherwise what this ends up as is quirky hooks that might work on cassandra as we manually index, but not on elasticsearch where we don't manually index the span names.

**Comment by drolando** [comment]
I closed this issue since I realized as you say that the best option is to
only put the http method in the client span name.

We always have server spans anyway so that name doesn't really matter. The
only car where there's no server is when we call external services like
Google or Amazon but those are custom use cases and it's up to the
developer to use a reasonable name.
